extensions-contrib/distinctcount/src/main/java/io/druid/query/aggregation/distinctcount/DistinctCountAggregatorFactory.java

```diff
@@ -0,0 +1,241 @@
+    this.bitMapFactory = bitMapFactory == null ? DEFAULT_BITMAP_FACTORY : bitMapFactory;
+  }
+
+  @Override
+  public Aggregator factorize(ColumnSelectorFactory columnFactory)
+  {
+    DimensionSelector selector = makeDimensionSelector(columnFactory);
+    if (selector == null) {
+      return new EmptyDistinctCountAggregator(name);
+    } else {
+      return new DistinctCountAggregator(
+          name,
+          selector,
+          bitMapFactory.makeEmptyMutableBitmap()
+      );
+    }
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory columnFactory)
+  {
+    DimensionSelector selector = makeDimensionSelector(columnFactory);
+    if (selector == null) {
+      return new EmptyDistinctCountBufferAggregator();
+    } else {
+      return new DistinctCountBufferAggregator(makeDimensionSelector(columnFactory));
+    }
+  }
+
+  private DimensionSelector makeDimensionSelector(final ColumnSelectorFactory columnFactory)
+  {
+    return columnFactory.makeDimensionSelector(new DefaultDimensionSpec(fieldName, fieldName));
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return new Comparator()
+    {
+      @Override
+      public int compare(Object o, Object o1)
+      {
+        return Longs.compare(((Number) o).longValue(), ((Number) o1).longValue());
+      }
+    };
+  }
+
+  @Override
+  public Object combine(Object lhs, Object rhs)
+  {
+    if (lhs == null && rhs == null) {
+      return 0L;
+    }
+    if (rhs == null) {
+      return ((Number) lhs).longValue();
+    }
+    if (lhs == null) {
+      return ((Number) rhs).longValue();
+    }
+    return ((Number) lhs).longValue() + ((Number) rhs).longValue();
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new LongSumAggregatorFactory(name, name);
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Arrays.<AggregatorFactory>asList(new DistinctCountAggregatorFactory(fieldName, fieldName, bitMapFactory));
+  }
+
+  @Override
+  public Object deserialize(Object object)
+  {
+    return object;
+  }
+
+  @Override
+  public Object finalizeComputation(Object object)
+  {
+    return object;
+  }
+
+  @JsonProperty
+  public String getFieldName()
+  {
+    return fieldName;
+  }
+
+  @JsonProperty("bitmap")
+  public BitMapFactory getBitMapFactory()
+  {
+    return bitMapFactory;
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Arrays.asList(fieldName);
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    byte[] fieldNameBytes = StringUtils.toUtf8(fieldName);
+    byte[] bitMapFactoryCacheKey = StringUtils.toUtf8(bitMapFactory.toString());
+    return ByteBuffer.allocate(1 + fieldNameBytes.length + bitMapFactoryCacheKey.length)
+                     .put(CACHE_TYPE_ID)
+                     .put(fieldNameBytes)
```

Comment:
`io.druid.query.aggregation.cardinality.CardinalityAggregatorFactory#getCacheKey` uses similar approach but proposed is fine for here.